servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultHttpServerBuilder.java

```diff
@@ -257,7 +256,8 @@ public HttpServerBuilder bufferAllocator(final BufferAllocator allocator) {
 
     @Override
     public Single<HttpServerContext> listen(final HttpService service) {
-        return listenForAdapter(toStreamingHttpService(service, computeServiceStrategy(HttpService.class, service)));
+        HttpExecutionStrategy executionStrategy = computeServiceStrategy(HttpService.class, service);
+        return listenForService(toStreamingHttpService(executionStrategy, service), executionStrategy);
```

Comment:
I'm puzzled why none of the tests caught this, but this is incorrect. Previous code of `listenForAdapter` used `adapterHolder.serviceInvocationStrategy()` which could be modified further by the conversion layer:
```java
    ServiceToStreamingService(final HttpService original, HttpExecutionStrategy strategy) {
        super(defaultStrategy() == strategy ? DEFAULT_STRATEGY : strategy);
        this.original = requireNonNull(original);
    }
```
Going forward, we need to use `requireOffloads()` method of the converted `StreamingHttpService` instead of `serviceInvocationStrategy()`.

Reply:
The name is confusing, but `defaultStrategy() != DEFAULT_STRATEGY ` 🤯

Reply:
I think I found the root cause of the issue: https://github.com/apple/servicetalk/pull/2460/files#diff-e1735bd3a1c16ec930f09a406521ed467238de450ea9a921adb469741ffd651bR39

The `ServiceAdapterHolder` did not override the `requiredOffloads()` so it gave conflicting strategies when asked for it via `requiredOffloads ` or `serviceInvocationStrategy `. If not overridden, it would always return from the parent:

```
default HttpExecutionStrategy requiredOffloads() {
        // safe default--implementations are expected to override
        return HttpExecutionStrategies.offloadAll();
    }
```

Reply:
Correct, we have to override `AbstractServiceAdapterHolder.requiredOffloads()` and use it instead of `serviceInvocationStrategy()`. That's one part.

The second part is `listenForAdapter` replacement. The current changes are not identical to how it behaved before. `listenForAdapter` used `adapterHolder.serviceInvocationStrategy()` for the 2nd argument of `listenForService`, but now you use `computeServiceStrategy`, which is not the same.
Consider removing `HttpExecutionStrategy` as an argument for `listenForService`. Instead, do this as the first line inside `listenForService`:
```java
final HttpExecutionStrategy computedStrategy = rawService.requiredOffloads();
```

Reply:
Applied with latest commit.